### PR TITLE
spotify-cli-linux: init at 1.6.0

### PR DESCRIPTION
**Pull request by kmein**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/lyricwikia/default.nix [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  propagatedBuildInputs = [ six beautifulsoup4 requests ];
+  doCheck = false;
+  meta = with stdenv.lib; {
+    homepage = "https://github.com/enricobacis/lyricwikia";
+    maintainers = [ maintainers.kmein ];
+    description = "LyricWikia API for song lyrics";
+    license = licenses.mit;
+    platforms = platforms.all;
+  };
+}
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/lyricwikia/default.nix:19]:
```suggestion
{ stdenv, fetchPypi, buildPythonPackage, pytestrunner, six, beautifulsoup4, requests }:
buildPythonPackage rec {
  pname = "lyricwikia";
  version = "0.1.11";

  src = fetchPypi {
    inherit pname version;
    sha256 = "0l5lkvr3299x79i7skdiggp67rzgax3s00psd1zqkxfysq27jvc8";
  };

  buildInputs = [ pytestrunner ];
  propagatedBuildInputs = [ six beautifulsoup4 requests ];
  doCheck = false;

  meta = with stdenv.lib; {
    homepage = "https://github.com/enricobacis/lyricwikia";
    maintainers = [ maintainers.kmein ];
    description = "LyricWikia API for song lyrics";
    license = licenses.mit;
    platforms = platforms.all;
  };
}
```

File: pkgs/development/python-modules/spotify-cli-linux/default.nix [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  propagatedBuildInputs = [ lyricwikia dbus-python ];
+  doCheck = false;
+  meta = with stdenv.lib; {
+    homepage = "https://pwittchen.github.io/spotify-cli-linux/";
+    maintainers = [ maintainers.kmein ];
+    description = "A command line interface to Spotify on Linux.";
+    license = licenses.gpl3;
+    platforms = platforms.linux;
+  };
+}
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/spotify-cli-linux/default.nix:18]:
```suggestion
{ stdenv, buildPythonApplication, fetchPypi, isPy3k, lyricwikia, dbus-python }:
buildPythonApplication rec {
  pname = "spotify-cli-linux";
  version = "1.6.0";

  src = fetchPypi {
    inherit pname version;
    sha256 = "0slyc3jfrj3rwq8rv6p5aqkw487aw7a87kmf1fb6n4vnvcf08v7w";
  };

  disabled = !isPy3k;
  propagatedBuildInputs = [ lyricwikia dbus-python ];
  doCheck = false;

  meta = with stdenv.lib; {
    homepage = "https://pwittchen.github.io/spotify-cli-linux/";
    maintainers = [ maintainers.kmein ];
    description = "A command line interface to Spotify on Linux.";
    license = licenses.gpl3;
    platforms = platforms.linux;
  };
}
```

Also only works on Python3.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 101547` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>python27Packages.spotify-cli-linux</li>
  </ul>
</details>
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>python27Packages.lyricwikia</li>
    <li>python37Packages.lyricwikia</li>
    <li>python37Packages.spotify-cli-linux</li>
    <li>python38Packages.lyricwikia</li>
    <li>python38Packages.spotify-cli-linux</li>
  </ul>
</details>

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -6732,6 +6734,8 @@ in {
 
   splinter = callPackage ../development/python-modules/splinter { };
 
+  spotify-cli-linux = callPackage ../development/python-modules/spotify-cli-linux { };
```
Comment (FRidh) [inline pkgs/top-level/python-packages.nix:6737]: This package set is for libraries primarily, a cli does not belong here and should move into the top-level

**Review by FRidh — CHANGES_REQUESTED** [review]
Why are the tests disabled? Include a comment in the expression explaining why the tests are disabled.

File: pkgs/top-level/python-packages.nix (hunk 3 above)
Comment (kmein) [reply pkgs/top-level/python-packages.nix:6737]: Ok, sorry, moving it over..

**Comment by kmein** [comment]
> Why are the tests disabled? Include a comment in the expression explaining why the tests are disabled.

I'm getting this error: `nix_run_setup: error: unrecognized arguments: test`

**Comment by SuperSandro2000** [comment]
Upstream only has linting and formatting checks but no actual code tests.

**Comment by berbiche** [comment]
There is a `dbus-send` call in the tool:
https://github.com/pwittchen/spotify-cli-linux/blob/b94bad8f4519c249909c7c9a176adfe2f789e2ca/spotifycli/spotifycli.py#L196-L200

I think it will need a `substituteInPlace` to have the right binary instead of using whatever `dbus-send` is available in the user environment.

File: pkgs/applications/audio/spotify-cli-linux/default.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    sha256 = "0slyc3jfrj3rwq8rv6p5aqkw487aw7a87kmf1fb6n4vnvcf08v7w";
+  };
+
+  preBuild = ''
+    substituteInPlace spotifycli/spotifycli.py \
+      --replace dbus-send ${dbus}/bin/dbus-send
+  '';
+
+  disabled = !python3Packages.isPy3k;
+  propagatedBuildInputs = [ python3Packages.lyricwikia python3Packages.dbus-python ];
```
Comment (sikmir) [inline pkgs/applications/audio/spotify-cli-linux/default.nix:17]:
```suggestion
  propagatedBuildInputs = with python3Packages; [ lyricwikia dbus-python ];
```

File: pkgs/development/python-modules/spotify-cli-linux/default.nix [hunk 5]
```diff
@@ -0,0 +1,22 @@
+{ stdenv, buildPythonApplication, fetchPypi, isPy3k, lyricwikia, dbus-python }:
+buildPythonApplication rec {
+  pname = "spotify-cli-linux";
```
Comment (sikmir) [inline pkgs/development/python-modules/spotify-cli-linux/default.nix:3]: Why you need this file? It duplicates `pkgs/applications/audio/spotify-cli-linux/default.nix`.

Comment (kmein) [reply pkgs/development/python-modules/spotify-cli-linux/default.nix]: Happened accidentally. Removed in the most recent commit.

File: pkgs/development/python-modules/lyricwikia/default.nix [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  version = "0.1.11";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0l5lkvr3299x79i7skdiggp67rzgax3s00psd1zqkxfysq27jvc8";
+  };
+
+  buildInputs = [ pytestrunner ];
+  propagatedBuildInputs = [ six beautifulsoup4 requests ];
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/lyricwikia/default.nix:13]: Can you add a comment why those tests are disabled?

Comment (kmein) [reply pkgs/development/python-modules/lyricwikia/default.nix:14]: Did so in the most recent commit.

**Pull request merged**